frontend/packages/console-app/src/actions/modify-hpa.ts

```diff
@@ -18,14 +20,24 @@ const hasHPAs = (mapOfResources: RelatedResources) =>
 const hpaRoute = ({ metadata: { name, namespace } }: K8sResourceCommon, kind: K8sKind) =>
   `/workload-hpa/ns/${namespace}/${referenceForModel(kind)}/${name}`;
 
+const isHelmWorkload = (labels: { [key: string]: string }): boolean =>
+  labels?.['app.kubernetes.io/managed-by'] === 'Helm' ||
+  labels?.['heritage'] === 'Helm' ||
+  _.has(labels, 'helm.sh/chart');
+
 export const AddHorizontalPodAutoScaler: KebabAction = (
   kind: K8sKind,
   obj: K8sResourceCommon,
   resources: RelatedResources,
+  customData?: { [key: string]: any },
 ) => ({
   label: `Add ${HorizontalPodAutoscalerModel.label}`,
   href: hpaRoute(obj, kind),
-  hidden: hasHPAs(resources),
+  hidden:
+    hasHPAs(resources) ||
+    isHelmWorkload(obj.metadata.labels) ||
+    resources?.['isOperatorBackedService'] ||
+    isOperatorBackedService(obj, customData?.csvs),
```

Comment:
```suggestion
    isOperatorBackedWorkload(obj, resources, customData),
```
That way it's simply "`isHelmWorkload`" & "`isOperatorBackedWorkload`". Easier to read.